packages/pages/src/components/image/image.tsx

```diff
@@ -1,14 +1,34 @@
 import * as React from "react";
-import { useRef, useState, useEffect } from "react";
-import { ImageProps } from "./types";
-
+import { useEffect, useRef, useState } from "react";
+import { ImageProps, ImageLayout } from "./types";
+
+/**
+ * A component that renders an image. Here is an example of using the component with simple and
+ * complex image fields from Knowledge Graph:
+ * ```
+ * import { Image } from "@yext/pages/components";
+ *
+ * const template = ({ document }) => {
+ *   return (
+ *     <div>
+ *       <Image image={document.logo.image} />
+ *       <Image image={document.photoGallery[0].image} />
+ *     </div>
+ *   );
+ * }
+ * ```
+ *
+ * `layout`, `imgOverrides` and `style` can be used to specify the style and behavior of the image.
```

Comment:
I think this is better left to documenting the prop type itself, rather than the component.